Subject: [PATCH] Mobile phone support.Renaming and new directory with development documentation
---
Index: src/components/Header.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/Header.tsx b/src/components/Header.tsx
--- a/src/components/Header.tsx	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/components/Header.tsx	(date 1767796320283)
@@ -1,23 +1,44 @@
-import {Link} from "wouter";
-import {useAuthStore} from "../stores/auth.ts";
-import {ThemeSelector} from "./basic/ThemeSelector.tsx";
+import { Link } from "wouter";
+import { useAuthStore } from "../stores/auth.ts";
+import { ThemeSelector } from "./basic/ThemeSelector.tsx";
 
 export const Header = () => {
-    const auth = useAuthStore()
+  const auth = useAuthStore();
 
-    return <div className={"navbar w-full bg-base-300 p-2 flex flex-col md:flex-row justify-between"}>
-        <div className={"flex items-center gap-2 justify-center"}>
-            <img src="/favicon.svg" alt="Favicon" width="40"/>
-            <h1 className={"text-xl"}>Altverse</h1>
-            <ThemeSelector></ThemeSelector>
-        </div>
-        <ul className="menu md:menu-horizontal bg-base-200 text-xl">
-            <li><Link href={"/"}>About</Link></li>
-            <li><Link href={"/dashboard"}>Dashboard</Link></li>
-        </ul>
-        <div className={"flex text-xl"}>
-            {!auth.valid &&  <Link href={"/login"} className={"btn btn-primary btn-lg"}>Login</Link>}
-            {auth.valid &&  <button onClick={() => auth.logout()} className={"btn btn-secondary btn-lg"}>Logout</button>}
-        </div>
+  return (
+    <div
+      className={
+        "navbar w-full bg-base-300 p-2 flex flex-col md:flex-row justify-between"
+      }
+    >
+      <div className={"flex items-center gap-2 justify-center"}>
+        <img src="/favicon.svg" alt="Favicon" width="40" />
+        <h1 className={"text-xl"}>Altverse</h1>
+        <ThemeSelector></ThemeSelector>
+      </div>
+      <ul className="menu md:menu-horizontal bg-base-200 text-xl">
+        <li>
+          <Link href={"/"}>About</Link>
+        </li>
+        <li>
+          <Link href={"/dashboard"}>Dashboard</Link>
+        </li>
+      </ul>
+      <div className={"flex text-xl"}>
+        {!auth.valid && (
+          <Link href={"/login"} className={"btn btn-primary btn-lg"}>
+            Login
+          </Link>
+        )}
+        {auth.valid && (
+          <button
+            onClick={() => auth.logout()}
+            className={"btn btn-primary btn-lg"}
+          >
+            Logout
+          </button>
+        )}
+      </div>
     </div>
-}
\ No newline at end of file
+  );
+};
Index: src/pages/Dashboard.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/pages/Dashboard.tsx b/src/pages/Dashboard.tsx
--- a/src/pages/Dashboard.tsx	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/pages/Dashboard.tsx	(date 1767865645717)
@@ -1,20 +1,28 @@
-import {Sidebar} from "../components/Sidebar.tsx";
-import {Route} from "wouter";
-import {GamePlay} from "./dashboard/Play.tsx";
-import {GameProfile} from "./dashboard/Profile.tsx";
-import {Base} from "./dashboard/Base.tsx";
-import {DevLaunch} from "./dashboard/Launch.tsx";
+import { Sidebar } from "../components/Sidebar.tsx";
+import { Route } from "wouter";
+import { GamePlay } from "./dashboard/Play.tsx";
+import { GameProfile } from "./dashboard/Profile.tsx";
+import { Base } from "./dashboard/Base.tsx";
+import { DevLaunch } from "./dashboard/Launch.tsx";
+import { GameChangelog } from "./dashboard/Changelog.tsx";
 
 export const Dashboard = () => {
-    return <>
-        <div className={"w-full flex flex-1 gap-2 flex-col md:flex-row"}>
-                <Sidebar></Sidebar>
-                    <div className={"mockup-window border border-base-300 bg-base-300 pr-4 w-full flex flex-row justify-center"}>
-                        <Route path={"/dashboard"} component={Base}/>
-                        <Route path={"/dashboard/play"} component={GamePlay}/>
-                        <Route path={"/dashboard/profile"} component={GameProfile}/>
-                        <Route path={"/dashboard/launch"} component={DevLaunch}/>
-                    </div>
-        </div>
-          </>
-}
\ No newline at end of file
+  return (
+    <>
+      <div className={"w-full flex flex-1 gap-2 flex-col md:flex-row"}>
+        <Sidebar></Sidebar>
+        <div
+          className={
+            "mockup-window border border-base-300 bg-base-300 pr-4 w-full flex flex-row justify-center"
+          }
+        >
+          <Route path={"/dashboard"} component={Base} />
+          <Route path={"/dashboard/play"} component={GamePlay} />
+          <Route path={"/dashboard/profile"} component={GameProfile} />
+          <Route path={"/dashboard/changelog"} component={GameChangelog} />
+          <Route path={"/dashboard/launch"} component={DevLaunch} />
+        </div>
+      </div>
+    </>
+  );
+};
Index: public/favicon.svg
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/public/favicon.svg b/public/favicon.svg
--- a/public/favicon.svg	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/public/favicon.svg	(date 1767796413465)
@@ -1,5 +1,46 @@
-<svg width="168" height="168" viewBox="0 0 168 168" fill="none" xmlns="http://www.w3.org/2000/svg">
-<rect width="168" height="168" rx="20" fill="#002946"/>
-<path d="M53.6468 73.7812C54.5449 71.0172 58.4551 71.0172 59.3532 73.7812L66.2664 95.0578C66.668 96.2938 67.8198 97.1307 69.1195 97.1307H91.4911C94.3972 97.1307 95.6056 100.85 93.2544 102.558L75.1555 115.707C74.104 116.471 73.664 117.825 74.0657 119.062L80.9788 140.338C81.8769 143.102 78.7135 145.4 76.3623 143.692L58.2634 130.543C57.2119 129.779 55.7881 129.779 54.7366 130.543L36.6377 143.692C34.2865 145.4 31.1231 143.102 32.0212 140.338L38.9343 119.062C39.336 117.825 38.896 116.471 37.8445 115.707L19.7456 102.558C17.3944 100.85 18.6028 97.1307 21.5089 97.1307H43.8805C45.1801 97.1307 46.332 96.2938 46.7336 95.0578L53.6468 73.7812Z" fill="#FFD6A7"/>
-<path d="M83 75L137 21M68.5 66.5L106 11M94.5 89L153.5 45.5" stroke="#FFD6A7" stroke-width="7" stroke-linecap="round"/>
+<svg width="158" height="158" viewBox="0 0 158 158" fill="none" xmlns="http://www.w3.org/2000/svg">
+<g filter="url(#filter0_n_244_81)">
+<path d="M6.74958 111C17.077 134.283 38.3284 151.643 64 156.578C60.7225 147.942 61.5429 140.884 62.2841 134.507C63.3522 125.317 64.2559 117.542 52.5 108.5C35 93 24.2471 99.8326 6.74958 111Z" fill="#2D2D3F"/>
+<path d="M79 0C35.3695 0 0 35.3695 0 79C0 83.4279 0.364281 87.7706 1.06451 92C37.4572 94.9967 53.8135 79 72 47.5C79.5618 33.8526 89.2794 34.1853 99.449 34.5335C110.125 34.8991 121.299 35.2817 131 19.5259C117.106 7.36772 98.9132 0 79 0Z" fill="#2D2D3F"/>
+<path d="M111.5 151.027C138.92 138.635 158 111.046 158 79C158 69.7043 156.394 60.7836 153.446 52.5C151.577 89.8445 144.376 90.8006 125.911 93.2523C124.205 93.4788 122.403 93.7181 120.5 94C93.5 98 93 143.5 111.5 151.027Z" fill="#2D2D3F"/>
+<path d="M79 158C90.5849 158 101.587 155.506 111.5 151.027C93 143.5 93.5 98 120.5 94C122.403 93.7181 124.205 93.4788 125.911 93.2523C144.376 90.8006 151.577 89.8445 153.446 52.5C151.96 48.3271 150.134 44.316 148 40.5C148.894 62.8459 126.219 74.1346 106.361 84.0206L106.36 84.0211L106.358 84.0223C104 85.1959 101.683 86.3498 99.449 87.5C81 97 61.7734 139.5 79 158Z" fill="#5F5CFF"/>
+<path d="M72 47.5C53.8135 79 37.4572 94.9967 1.06451 92C2.16559 98.6505 4.09733 105.021 6.74958 111C24.2471 99.8326 35 93 52.5 108.5C64.2559 117.542 63.3522 125.317 62.2841 134.507C61.5429 140.884 60.7225 147.942 64 156.578C68.8565 157.511 73.8712 158 79 158C61.7734 139.5 81 97 99.449 87.5C101.683 86.3498 104 85.1959 106.358 84.0223L106.36 84.0211L106.361 84.0206C126.219 74.1346 148.894 62.8459 148 40.5C143.562 32.5638 137.795 25.472 131 19.5259C121.299 35.2817 110.125 34.8991 99.449 34.5335C89.2794 34.1853 79.5618 33.8526 72 47.5Z" fill="#A09EFF"/>
+</g>
+<path d="M19.5 63L21.2556 67.7444L26 69.5L21.2556 71.2556L19.5 76L17.7444 71.2556L13 69.5L17.7444 67.7444L19.5 63Z" fill="#E8E7FE"/>
+<path d="M20.7871 67.918L20.8662 68.1338L21.082 68.2129L24.5596 69.5L21.082 70.7871L20.8662 70.8662L20.7871 71.082L19.5 74.5596L18.2129 71.082L18.1338 70.8662L17.918 70.7871L14.4395 69.5L17.918 68.2129L18.1338 68.1338L18.2129 67.918L19.5 64.4395L20.7871 67.918Z" stroke="#E8E7FE" stroke-opacity="0.392157"/>
+<path d="M42.5 31L44.2556 35.7444L49 37.5L44.2556 39.2556L42.5 44L40.7444 39.2556L36 37.5L40.7444 35.7444L42.5 31Z" fill="#E8E7FE"/>
+<path d="M43.7871 35.918L43.8662 36.1338L44.082 36.2129L47.5596 37.5L44.082 38.7871L43.8662 38.8662L43.7871 39.082L42.5 42.5596L41.2129 39.082L41.1338 38.8662L40.918 38.7871L37.4395 37.5L40.918 36.2129L41.1338 36.1338L41.2129 35.918L42.5 32.4395L43.7871 35.918Z" stroke="#E8E7FE" stroke-opacity="0.392157"/>
+<path d="M93.5 14L95.2556 18.7444L100 20.5L95.2556 22.2556L93.5 27L91.7444 22.2556L87 20.5L91.7444 18.7444L93.5 14Z" fill="#E8E7FE"/>
+<path d="M94.7871 18.918L94.8662 19.1338L95.082 19.2129L98.5596 20.5L95.082 21.7871L94.8662 21.8662L94.7871 22.082L93.5 25.5596L92.2129 22.082L92.1338 21.8662L91.918 21.7871L88.4395 20.5L91.918 19.2129L92.1338 19.1338L92.2129 18.918L93.5 15.4395L94.7871 18.918Z" stroke="#E8E7FE" stroke-opacity="0.392157"/>
+<path d="M49.5 56L51.2556 60.7444L56 62.5L51.2556 64.2556L49.5 69L47.7444 64.2556L43 62.5L47.7444 60.7444L49.5 56Z" fill="#E8E7FE"/>
+<path d="M50.7871 60.918L50.8662 61.1338L51.082 61.2129L54.5596 62.5L51.082 63.7871L50.8662 63.8662L50.7871 64.082L49.5 67.5596L48.2129 64.082L48.1338 63.8662L47.918 63.7871L44.4395 62.5L47.918 61.2129L48.1338 61.1338L48.2129 60.918L49.5 57.4395L50.7871 60.918Z" stroke="#E8E7FE" stroke-opacity="0.392157"/>
+<path d="M32.5 109L34.2556 113.744L39 115.5L34.2556 117.256L32.5 122L30.7444 117.256L26 115.5L30.7444 113.744L32.5 109Z" fill="#E8E7FE"/>
+<path d="M33.7871 113.918L33.8662 114.134L34.082 114.213L37.5596 115.5L34.082 116.787L33.8662 116.866L33.7871 117.082L32.5 120.56L31.2129 117.082L31.1338 116.866L30.918 116.787L27.4395 115.5L30.918 114.213L31.1338 114.134L31.2129 113.918L32.5 110.439L33.7871 113.918Z" stroke="#E8E7FE" stroke-opacity="0.392157"/>
+<path d="M49.5 129L51.2556 133.744L56 135.5L51.2556 137.256L49.5 142L47.7444 137.256L43 135.5L47.7444 133.744L49.5 129Z" fill="#E8E7FE"/>
+<path d="M50.7871 133.918L50.8662 134.134L51.082 134.213L54.5596 135.5L51.082 136.787L50.8662 136.866L50.7871 137.082L49.5 140.56L48.2129 137.082L48.1338 136.866L47.918 136.787L44.4395 135.5L47.918 134.213L48.1338 134.134L48.2129 133.918L49.5 130.439L50.7871 133.918Z" stroke="#E8E7FE" stroke-opacity="0.392157"/>
+<path d="M141.5 96L143.256 100.744L148 102.5L143.256 104.256L141.5 109L139.744 104.256L135 102.5L139.744 100.744L141.5 96Z" fill="#E8E7FE"/>
+<path d="M142.787 100.918L142.866 101.134L143.082 101.213L146.56 102.5L143.082 103.787L142.866 103.866L142.787 104.082L141.5 107.56L140.213 104.082L140.134 103.866L139.918 103.787L136.439 102.5L139.918 101.213L140.134 101.134L140.213 100.918L141.5 97.4395L142.787 100.918Z" stroke="#E8E7FE" stroke-opacity="0.392157"/>
+<path d="M117.5 118L119.256 122.744L124 124.5L119.256 126.256L117.5 131L115.744 126.256L111 124.5L115.744 122.744L117.5 118Z" fill="#E8E7FE"/>
+<path d="M118.787 122.918L118.866 123.134L119.082 123.213L122.56 124.5L119.082 125.787L118.866 125.866L118.787 126.082L117.5 129.56L116.213 126.082L116.134 125.866L115.918 125.787L112.439 124.5L115.918 123.213L116.134 123.134L116.213 122.918L117.5 119.439L118.787 122.918Z" stroke="#E8E7FE" stroke-opacity="0.392157"/>
+<path d="M62.5 12L64.2556 16.7444L69 18.5L64.2556 20.2556L62.5 25L60.7444 20.2556L56 18.5L60.7444 16.7444L62.5 12Z" fill="#E8E7FE"/>
+<path d="M63.7871 16.918L63.8662 17.1338L64.082 17.2129L67.5596 18.5L64.082 19.7871L63.8662 19.8662L63.7871 20.082L62.5 23.5596L61.2129 20.082L61.1338 19.8662L60.918 19.7871L57.4395 18.5L60.918 17.2129L61.1338 17.1338L61.2129 16.918L62.5 13.4395L63.7871 16.918Z" stroke="#E8E7FE" stroke-opacity="0.392157"/>
+<path d="M63 26.5L53 57.5M42 66L28 69.5M21.5 62.5C21.9 62.1 33 48.6667 38.5 42M47.5 33.5L59 23M70.5 18.5C71.7 18.5 80.6667 19.8333 85 20.5M37.5 121L45.5 131M124 120L137 108" stroke="#E8E7FE" stroke-opacity="0.392157" stroke-width="2"/>
+<defs>
+<filter id="filter0_n_244_81" x="0" y="0" width="158" height="158" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB">
+<feFlood flood-opacity="0" result="BackgroundImageFix"/>
+<feBlend mode="normal" in="SourceGraphic" in2="BackgroundImageFix" result="shape"/>
+<feTurbulence type="fractalNoise" baseFrequency="2 2" stitchTiles="stitch" numOctaves="3" result="noise" seed="3653" />
+<feColorMatrix in="noise" type="luminanceToAlpha" result="alphaNoise" />
+<feComponentTransfer in="alphaNoise" result="coloredNoise1">
+<feFuncA type="discrete" tableValues="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 "/>
+</feComponentTransfer>
+<feComposite operator="in" in2="shape" in="coloredNoise1" result="noise1Clipped" />
+<feFlood flood-color="rgba(0, 0, 0, 0.25)" result="color1Flood" />
+<feComposite operator="in" in2="noise1Clipped" in="color1Flood" result="color1" />
+<feMerge result="effect1_noise_244_81">
+<feMergeNode in="shape" />
+<feMergeNode in="color1" />
+</feMerge>
+</filter>
+</defs>
 </svg>
Index: src/components/basic/ThemeSelector.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/basic/ThemeSelector.tsx b/src/components/basic/ThemeSelector.tsx
--- a/src/components/basic/ThemeSelector.tsx	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/components/basic/ThemeSelector.tsx	(date 1767023055834)
@@ -4,9 +4,11 @@
     const theme= useThemeStore()
 
     return <div className="dropdown dropdown-">
+        {/* @ts-ignore */}
         <div tabIndex="0" role="button" className="btn m-1 btn-lg">
             Theme
         </div>
+        {/* @ts-ignore */}
         <ul tabIndex="-1" className="dropdown-content bg-base-300 rounded-box z-1 w-52 p-2">
             {ThemesNames.map((v, i) => <li>
                 <input
Index: package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/package.json b/package.json
--- a/package.json	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/package.json	(date 1767279172738)
@@ -1,39 +1,40 @@
 {
-	"name": "client",
-	"private": true,
-	"version": "0.0.0",
-	"type": "module",
-	"scripts": {
-		"dev": "vite --port 7010",
-		"build": "tsc -b && vite build",
-		"preview": "vite preview"
-	},
-	"dependencies": {
-		"@types/snappyjs": "^0.7.1",
-		"bootstrap-icons": "^1.13.1",
-		"js-cookie": "^3.0.5",
-		"lz4js": "^0.2.0",
-		"mitt": "^3.0.1",
-		"preact": "^10.27.2",
-		"protobufjs": "^7.5.4",
-		"snappyjs": "^0.7.0",
-		"wouter": "^3.8.1",
-		"zustand": "^5.0.9"
-	},
-	"devDependencies": {
-		"@grpc/grpc-js": "^1.14.3",
-		"@grpc/proto-loader": "^0.8.0",
-		"@preact/preset-vite": "^2.10.2",
-		"@rollup/plugin-alias": "^6.0.0",
-		"@tailwindcss/vite": "^4.1.17",
-		"@types/js-cookie": "^3",
-		"@types/lz4js": "^0.2.2",
-		"@types/node": "^24.10.1",
-		"daisyui": "^5.5.14",
-		"tailwindcss": "^4.1.17",
-		"typescript": "~5.9.3",
-		"vite": "^7.2.4",
-		"vite-plugin-static-copy": "^3.1.4",
-		"vite-plugin-wasm": "^3.5.0"
-	}
+  "name": "client",
+  "private": true,
+  "version": "0.0.0",
+  "type": "module",
+  "scripts": {
+    "dev": "vite --port 7010",
+    "build": "tsc -b && vite build",
+    "preview": "vite preview"
+  },
+  "dependencies": {
+    "@types/snappyjs": "^0.7.1",
+    "bootstrap-icons": "^1.13.1",
+    "js-cookie": "^3.0.5",
+    "lz4js": "^0.2.0",
+    "mitt": "^3.0.1",
+    "preact": "^10.27.2",
+    "protobufjs": "^7.5.4",
+    "snappyjs": "^0.7.0",
+    "wouter": "^3.8.1",
+    "zustand": "^5.0.9"
+  },
+  "devDependencies": {
+    "@grpc/grpc-js": "^1.14.3",
+    "@grpc/proto-loader": "^0.8.0",
+    "@preact/preset-vite": "^2.10.2",
+    "@rollup/plugin-alias": "^6.0.0",
+    "@tailwindcss/vite": "^4.1.17",
+    "@types/js-cookie": "^3",
+    "@types/lz4js": "^0.2.2",
+    "@types/node": "^24.10.1",
+    "daisyui": "^5.5.14",
+    "prettier": "^3.7.4",
+    "tailwindcss": "^4.1.17",
+    "typescript": "~5.9.3",
+    "vite": "^7.2.4",
+    "vite-plugin-static-copy": "^3.1.4",
+    "vite-plugin-wasm": "^3.5.0"
+  }
 }
Index: src/components/Profile.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/Profile.tsx b/src/components/Profile.tsx
--- a/src/components/Profile.tsx	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/components/Profile.tsx	(date 1767294359091)
@@ -1,63 +1,70 @@
-import {Card} from "./basic/Card.tsx";
-import {useEffect, useRef, useState} from "preact/hooks";
-import {ApiRequests} from "../api/requests.ts";
-import {useAuthStore} from "../stores/auth.ts";
-import {type Profile, ResponseMessage} from "../api/types.ts";
-import {AccountRole} from "../types.ts";
+import { Card } from "./basic/Card.tsx";
+import { useEffect, useRef, useState } from "preact/hooks";
+import { ApiRequests } from "../api/requests.ts";
+import { useAuthStore } from "../stores/auth.ts";
+import { type Profile, ResponseMessage } from "../api/types.ts";
+import { AccountRole } from "../types.ts";
 
 interface Props {
-    username?: string;
+  username?: string;
 }
 
 export const ProfileCard = (props: Props) => {
-    const [profile, setProfile] = useState<Profile>()
-    const [error, setError] = useState(false)
-    const timeout = useRef<NodeJS.Timeout>(0)
-    useEffect(() => {
-        const fetch = async () => {
-            if (props.username) {
-                const value = await ApiRequests.profile(props.username)
-                if (value.status !== ResponseMessage.NotFound) {
-                    setProfile(value.profile)
-                    console.log(value.profile)
-                    setError(false)
-                } else {
-                    setError(true)
-                }
-            }
-            else setProfile(useAuthStore.getState().profile!)
+  const [profile, setProfile] = useState<Profile>();
+  const [error, setError] = useState(false);
+  const timeout = useRef<NodeJS.Timeout>();
+  useEffect(() => {
+    const fetch = async () => {
+      if (props.username) {
+        const value = await ApiRequests.profile(props.username);
+        if (value.status !== ResponseMessage.NotFound) {
+          setProfile(value.profile);
+          console.log(value.profile);
+          setError(false);
+        } else {
+          setError(true);
         }
-        clearTimeout(timeout.current)
-        timeout.current = setTimeout(fetch, 200)
-    }, [props.username])
+      } else setProfile(useAuthStore.getState().profile!);
+    };
+    clearTimeout(timeout.current);
+    timeout.current = setTimeout(fetch, 200);
+  }, [props.username]);
 
-    const role = () => {
-        switch (profile?.role) {
-            case AccountRole.Dev:
-                return <div className={"badge badge-primary text-xl"}>Dev</div>
-            case AccountRole.Mod:
-                return <div className={"badge badge-secondary text-xl"}>Mod</div>
-        }
+  const role = () => {
+    switch (profile?.role) {
+      case AccountRole.Dev:
+        return <div className={"badge badge-primary text-xl"}>Dev</div>;
+      case AccountRole.Mod:
+        return <div className={"badge badge-secondary text-xl"}>Mod</div>;
     }
+  };
 
-    if (error)
-        return <div className={'w-full alert alert-error alert-soft text-xl'}>Profile is not found
-        </div>
+  if (error)
+    return (
+      <div className={"w-full alert alert-error alert-soft text-xl"}>
+        Profile is not found
+      </div>
+    );
 
-    return <div className={'w-full'}><Card size={"sm"} loading={profile == undefined}>
+  return (
+    <div className={"w-full"}>
+      <Card size={"sm"} loading={profile == undefined}>
         <div className={"flex flex-col"}>
-        <h1 className={"text-2xl flex flex-row  items-center gap-2"}>
+          <h1 className={"text-2xl flex flex-row  items-center gap-2"}>
             {profile?.username}
             {role()}
-        </h1>
-        <div className={"text-lg"}>
-            {profile != undefined && <>Victory Points: {profile?.vp}</> }
-        </div>
-            {profile != undefined && Object.keys(profile?.highest).length !== 0 &&
-            <details>
+          </h1>
+          <div className={"text-lg"}>
+            {profile != undefined && <>Victory Points: {profile?.vp}</>}
+          </div>
+          {profile != undefined &&
+            Object.keys(profile?.highest).length !== 0 && (
+              <details>
                 <summary className={"text-xl"}>Highest</summary>
-            </details>}
+              </details>
+            )}
         </div>
-    </Card>
+      </Card>
     </div>
-}
\ No newline at end of file
+  );
+};
Index: src/index.css
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/index.css b/src/index.css
--- a/src/index.css	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/index.css	(date 1767883025763)
@@ -1,14 +1,28 @@
 @import "tailwindcss";
-@import 'bootstrap-icons/font/bootstrap-icons.css';
+@import "bootstrap-icons/font/bootstrap-icons.css";
 @import "./themes.css";
 
 @plugin "daisyui" {
-    themes: light-plus --default, dark-plus --prefersdark;
-    root: ":root";
+  themes:
+    light-plus --default,
+    dark-plus --prefersdark;
+  root: ":root";
 }
 
 :root {
-    font-family: "Kanit";
+  font-family: "Kanit", sans-serif;
+}
+
+.animate-sbounce {
+  animation: sbounce 4s linear alternate infinite;
+}
+@keyframes sbounce {
+  100% {
+    transform: translateY(-10px);
+  }
+  0% {
+    transform: translateY(10px);
+  }
 }
 /*    font-weight: 500;*/
 
Index: src/pages/dashboard/Base.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/pages/dashboard/Base.tsx b/src/pages/dashboard/Base.tsx
--- a/src/pages/dashboard/Base.tsx	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/pages/dashboard/Base.tsx	(date 1767807785117)
@@ -1,11 +1,39 @@
 export const Base = () => {
-    return <div className={"md:w-[700px] w-full text-2xl flex flex-col gap-2 pt-4"}>
-        <h1 className={"text-center text-3xl"}>Welcome to Altverse! Have fun!</h1>
-        <p className={"alert alert-warning alert-soft text-2xl"}>Due to the state of the game, registration of new accounts is done using a token.</p>
-        <h1 className={"divider text-3xl"}>A little about the game</h1>
-        <p>The game is inspired by another io game, the original game is called <a href="https://evades,io" target={"_blank"} className={"link link-info"}>evades.io</a>. </p>
-        <p>It is currently being slowly developed by <a href="https://ether.cd" target={"_blank"} className={"link link-info"}>EtherCD</a>. </p>
-        <h1 className={"divider text-3xl"}>Game idea</h1>
-        <p>The original idea of ​​the game was to make an alternative version of the original, with new content and features.</p>
+  return (
+    <div className={"md:w-[700px] w-full text-xl flex flex-col gap-2 pt-4"}>
+      <h1 className={"text-center"}>Welcome to Altverse! Have fun!</h1>
+      <p className={"alert alert-warning alert-soft text-xl"}>
+        Due to the state of the game, registration of new accounts is done using
+        a token.
+      </p>
+      <h1 className={"divider text-3xl"}>A little about the game</h1>
+      <p>
+        The game is inspired by another io game, the original game is called{" "}
+        <a
+          href="https://evades.io"
+          target={"_blank"}
+          className={"link link-info"}
+        >
+          evades.io
+        </a>
+        .{" "}
+      </p>
+      <p>
+        It is currently being slowly developed by{" "}
+        <a
+          href="https://ether.cd"
+          target={"_blank"}
+          className={"link link-info"}
+        >
+          EtherCD
+        </a>
+        .{" "}
+      </p>
+      <h1 className={"divider text-3xl"}>Game idea</h1>
+      <p>
+        The original idea of ​​the game was to make an alternative version of
+        the original, with new content and features.
+      </p>
     </div>
-}
\ No newline at end of file
+  );
+};
Index: README.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/README.md b/README.md
--- a/README.md	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/README.md	(date 1767802331901)
@@ -1,5 +1,5 @@
 <div align="center">
-  <img src="./misc/logotype.svg" />
+  <img src="./public/favicon.svg" />
+  <h1>Altverse</h1>
   <p>A simple game/engine, a parody of evades.io</p>
 </div>
-
Index: src/components/game/index.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/game/index.tsx b/src/components/game/index.tsx
--- a/src/components/game/index.tsx	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/components/game/index.tsx	(date 1767045445968)
@@ -12,11 +12,13 @@
   const [canvasRef, reason] = useGame(props.ip);
 
   return (
-    <div className={style.game}>
-      <canvas width={1280} height={720} ref={canvasRef}></canvas>
-      <LeaderBoard></LeaderBoard>
-      <Chat></Chat>
-      <Close reason={reason ?? ""} />
+    <div className={"transform"}>
+      <div className={style.game}>
+        <canvas width={1280} height={720} ref={canvasRef}></canvas>
+        <LeaderBoard></LeaderBoard>
+        <Chat></Chat>
+        <Close reason={reason ?? ""} />
+      </div>
     </div>
   );
 }
Index: src/pages/dashboard/Changelog.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/pages/dashboard/Changelog.tsx b/src/pages/dashboard/Changelog.tsx
new file mode 100644
--- /dev/null	(date 1767877183579)
+++ b/src/pages/dashboard/Changelog.tsx	(date 1767877183579)
@@ -0,0 +1,65 @@
+import { useMemo, useRef, useState } from "preact/hooks";
+
+export const GameChangelog = () => {
+  const modalRef = useRef<HTMLDialogElement>(null);
+  const [choosed, setChoosed] = useState<number>(0);
+
+  const changelog = [
+    {
+      title: "Small Update",
+      features: ["optimised working with"],
+      date: "08.01.26",
+    },
+  ];
+
+  const change = useMemo(() => {
+    return changelog[choosed];
+  }, [choosed]);
+
+  return (
+    <div className={"w-full"}>
+      <h1 className={"divider text-2xl"}>Changelog</h1>
+      <div className={"grid grid-cols-3 gap-4"}>
+        {changelog.map((v, i) => (
+          <div
+            className={
+              "card bg-base-100 min-h-60 p-5 flex flex-col  max-h-50 relative"
+            }
+          >
+            <h1 className={"text-xl text-center"}>{v.title}</h1>
+            <h2 className={"text-xl text-center"}>{v.date}</h2>
+            <div className={"from-base-100 to-base-300 h-60 absolute"}></div>
+            <ul className={"flex-1 overflow-hidden list-decimal"}>
+              {v.features.map((feature) => (
+                <li>{feature}</li>
+              ))}
+            </ul>
+            <button
+              className={"btn"}
+              onClick={() => {
+                modalRef.current?.showModal();
+                setChoosed(i);
+              }}
+            >
+              See full
+            </button>
+          </div>
+        ))}
+      </div>
+      <dialog className="modal" ref={modalRef}>
+        <div className="modal-box">
+          <h1 className={"text-xl text-center"}>{change.title}</h1>
+          <h2 className={"text-xl text-center"}>{change.date}</h2>
+          <ul className={"flex-1 overflow-hidden list-disc"}>
+            {change.features.map((feature) => (
+              <li>{feature}</li>
+            ))}
+          </ul>
+        </div>
+        <form method="dialog" className="modal-backdrop">
+          <button>close</button>
+        </form>
+      </dialog>
+    </div>
+  );
+};
Index: bun.lock
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bun.lock b/bun.lock
--- a/bun.lock	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/bun.lock	(date 1767045297453)
@@ -26,6 +26,7 @@
         "@types/lz4js": "^0.2.2",
         "@types/node": "^24.10.1",
         "daisyui": "^5.5.14",
+        "prettier": "^3.7.4",
         "tailwindcss": "^4.1.17",
         "typescript": "~5.9.3",
         "vite": "^7.2.4",
@@ -423,6 +424,8 @@
 
     "preact": ["preact@10.28.1", "", {}, "sha512-u1/ixq/lVQI0CakKNvLDEcW5zfCjUQfZdK9qqWuIJtsezuyG6pk9TWj75GMuI/EzRSZB/VAE43sNWWZfiy8psw=="],
 
+    "prettier": ["prettier@3.7.4", "", { "bin": { "prettier": "bin/prettier.cjs" } }, "sha512-v6UNi1+3hSlVvv8fSaoUbggEM5VErKmmpGA7Pl3HF8V6uKY7rvClBOJlH6yNwQtfTueNkGVpOv/mtWL9L4bgRA=="],
+
     "protobufjs": ["protobufjs@7.5.4", "", { "dependencies": { "@protobufjs/aspromise": "^1.1.2", "@protobufjs/base64": "^1.1.2", "@protobufjs/codegen": "^2.0.4", "@protobufjs/eventemitter": "^1.1.0", "@protobufjs/fetch": "^1.1.0", "@protobufjs/float": "^1.0.2", "@protobufjs/inquire": "^1.1.0", "@protobufjs/path": "^1.1.2", "@protobufjs/pool": "^1.1.0", "@protobufjs/utf8": "^1.1.0", "@types/node": ">=13.7.0", "long": "^5.0.0" } }, "sha512-CvexbZtbov6jW2eXAvLukXjXUW1TzFaivC46BpWc/3BpcCysb5Vffu+B3XHMm8lVEuy2Mm4XGex8hBSg1yapPg=="],
 
     "react": ["react@19.2.3", "", {}, "sha512-Ku/hhYbVjOQnXDZFv2+RibmLFGwFdeeKHFcOTlrt7xplBnya5OGn/hIRDsqDiSUcfORsDC7MPxwork8jBwsIWA=="],
Index: src/components/Sidebar.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/Sidebar.tsx b/src/components/Sidebar.tsx
--- a/src/components/Sidebar.tsx	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/components/Sidebar.tsx	(date 1767953271577)
@@ -1,47 +1,75 @@
-import {Link} from "wouter";
-import {useAuthStore} from "../stores/auth.ts";
+import { Link } from "wouter";
+import { useAuthStore } from "../stores/auth.ts";
 
 export const Sidebar = () => {
-    const auth = useAuthStore()
+  const auth = useAuthStore();
 
-    return <div className={"flex p-2 bg-base-300 flex-col md:w-[350px]"}>
-        <ul className="menu rounded-box w-full text-xl">
-            {auth.valid && <>
-                <li>
-                    <h2 className="menu-title flex gap-2 text-xl"><i className="bi bi-controller"></i> Game</h2>
-                    <ul>
-                        <li><Link href={"/dashboard/play"}>Play</Link></li>
-                        <li><Link href={"/dashboard/profile"}>Profiles</Link></li>
-                    </ul>
+  return (
+    <div className={"flex p-2 bg-base-300 flex-col md:w-[350px]"}>
+      <ul className="menu rounded-box w-full text-xl">
+        {auth.valid && (
+          <>
+            <li>
+              <h2 className="menu-title flex gap-2 text-xl">
+                <i className="bi bi-controller"></i> Game
+              </h2>
+              <ul>
+                <li>
+                  <Link href={"/dashboard/play"}>Play</Link>
+                </li>
+                <li>
+                  <Link href={"/dashboard/profile"}>Profiles</Link>
                 </li>
                 {/*<li>*/}
-                {/*    <h2 className="menu-title flex gap-2 text-xl"><i className="bi bi-person"></i>Account</h2>*/}
-                {/*    <ul>*/}
-                {/*        <li><Link href={"/dashboard"}>Settings</Link></li>*/}
-                {/*    </ul>*/}
+                {/*  <Link href={"/dashboard/changelog"}>Changelog</Link>*/}
                 {/*</li>*/}
-            </>}
+              </ul>
+            </li>
             <li>
-                <h2 className="menu-title flex gap-2 text-xl"><i className="bi bi-person"></i>Development</h2>
-                <ul>
-                    <li><Link href={"/dashboard/launch"}>Launch</Link></li>
-                </ul>
-            </li>
-
-        </ul>
+              <h2 className="menu-title flex gap-2 text-xl">
+                <i className="bi bi-person"></i>Account
+              </h2>
+              <ul>
+                <li>
+                  <Link href={"/dashboard/settings"}>Settings</Link>
+                </li>
+              </ul>
+            </li>
+          </>
+        )}
+        <li>
+          <h2 className="menu-title flex gap-2 text-xl">
+            <i class="bi bi-gear-fill"></i>Development
+          </h2>
+          <ul>
+            <li>
+              <Link href={"/dashboard/launch"}>Launch</Link>
+            </li>
+          </ul>
+        </li>
+      </ul>
     </div>
-}
+  );
+};
 
 export const DocsSidebar = () => {
-    return <div className={"flex p-2 bg-base-300 flex-col w-[350px]"}>
-        <ul className="menu rounded-box w-full text-xl">
-            <li>
-                <h2 className="menu-title flex gap-2 text-xl"><i className="bi bi-person"></i>Deploy</h2>
-                <ul>
-                    <li><Link href={"/home"}>Dev mode</Link></li>
-                    <li><Link href={"/home/profile"}>Nginx configuration</Link></li>
-                </ul>
-            </li>
-        </ul>
+  return (
+    <div className={"flex p-2 bg-base-300 flex-col w-[350px]"}>
+      <ul className="menu rounded-box w-full text-xl">
+        <li>
+          <h2 className="menu-title flex gap-2 text-xl">
+            <i className="bi bi-person"></i>Deploy
+          </h2>
+          <ul>
+            <li>
+              <Link href={"/home"}>Dev mode</Link>
+            </li>
+            <li>
+              <Link href={"/home/profile"}>Nginx configuration</Link>
+            </li>
+          </ul>
+        </li>
+      </ul>
     </div>
-}
\ No newline at end of file
+  );
+};
Index: src/pages/About.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/pages/About.tsx b/src/pages/About.tsx
--- a/src/pages/About.tsx	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/pages/About.tsx	(date 1767810836054)
@@ -1,11 +1,12 @@
-import {Card} from "../components/basic/Card.tsx";
-import {Link} from "wouter";
-import {useAuthStore} from "../stores/auth.ts";
+import { Card } from "../components/basic/Card.tsx";
+import { Link } from "wouter";
+import { useAuthStore } from "../stores/auth.ts";
 
 export const About = () => {
   const auth = useAuthStore();
 
-  return <>
+  return (
+    <>
       <div className={"flex gap-2 flex-col md:flex-row"}>
         <Card>
           <h1 className={"text-xl"}>Game status</h1>
@@ -18,40 +19,69 @@
           </ul>
         </Card>
         <div
-            className={"card text-center md:text-left bg-base-300 md:w-[800px] w-full flex flex-col md:flex-row items-center md:gap-20 gap-2 justify-center"}>
+          className={
+            "card text-center md:text-left bg-[radial-gradient(circle_at_75%_50%,#8b5cf620,#00000000)] md:w-[800px] w-full flex flex-col md:flex-row items-center md:gap-20 gap-2 justify-center"
+          }
+        >
           <div className={"flex flex-col gap-2"}>
             <h1 className={"text-4xl"}>Altverse</h1>
-          <p className={"text-2xl"}>A fan game based on evades.io.</p>
-          <div className={"alert alert-warning alert-soft text-xl"}>
-            Game still in deep developing
-          </div>
-          {auth.valid &&
-              <Link href={"/home"} className={"btn btn-lg btn-accent"}>Play now!</Link>
-          }
-          {!auth.valid &&
-              <Link href={"/login"} className={"btn btn-lg btn-accent"}>Sign in to play</Link>
-          }
-        </div>
-        <img src="/favicon.svg" alt="Favicon" width={190}/>
-      </div>
+            <p className={"text-2xl"}>A fan game based on evades.io.</p>
+            <div className={"alert alert-warning alert-soft text-xl"}>
+              Game still in deep developing
+            </div>
+            {auth.valid && (
+              <Link
+                href={"/dashboard/play"}
+                className={"btn btn-lg btn-primary"}
+              >
+                Play now!
+              </Link>
+            )}
+            {!auth.valid && (
+              <Link href={"/login"} className={"btn btn-lg btn-primary"}>
+                Sign in to play
+              </Link>
+            )}
+          </div>
+          <img
+            src="/favicon.svg"
+            alt="Favicon"
+            width={190}
+            className={"animate-sbounce"}
+          />
+        </div>
       </div>
       <div className="divider text-2xl">Key features</div>
       <h1 className={"text-center text-2xl"}></h1>
       <div className={"flex gap-2 flex-col md:flex-row"}>
         <Card>
-          <h1 className={"text-xl flex gap-1"}><i class="bi bi-hdd-stack-fill"></i>Scalability</h1>
-          <p className={"text-lg"}>By connecting the main server to the game servers using gRPC, the game can easily scale to any number of
-            servers.</p>
+          <h1 className={"text-xl flex gap-1"}>
+            <i class="bi bi-hdd-stack-fill"></i>Scalability
+          </h1>
+          <p className={"text-lg"}>
+            By connecting the main server to the game servers using gRPC, the
+            game can easily scale to any number of servers.
+          </p>
         </Card>
         <Card>
-          <h1 className={"text-xl flex gap-1"}><i class="bi bi-git"></i>OpenSource</h1>
-          <p className={"text-lg"}>The game code is available on GitHub and is open source, licensed under the GPL-3 license.</p>
+          <h1 className={"text-xl flex gap-1"}>
+            <i class="bi bi-git"></i>OpenSource
+          </h1>
+          <p className={"text-lg"}>
+            The game code is available on GitHub and is open source, licensed
+            under the GPL-3 license.
+          </p>
         </Card>
         <Card>
-          <h1 className={"text-xl flex gap-1"}><i class="bi bi-lightning-charge-fill"></i>Performance</h1>
-          <p className={"text-lg"}>The game server consists of NodeJS and Rust game logic, using optimized and efficient libraries, ensuring
-            high performance.</p>
+          <h1 className={"text-xl flex gap-1"}>
+            <i class="bi bi-lightning-charge-fill"></i>Performance
+          </h1>
+          <p className={"text-lg"}>
+            The game server consists of NodeJS and Rust game logic, using
+            optimized and efficient libraries, ensuring high performance.
+          </p>
         </Card>
       </div>
-  </>
+    </>
+  );
 };
Index: src/components/game/index.module.css
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/game/index.module.css b/src/components/game/index.module.css
--- a/src/components/game/index.module.css	(revision 44dca8d342b670a4e14268e6f6655030b61ad8e5)
+++ b/src/components/game/index.module.css	(date 1767040614093)
@@ -2,8 +2,8 @@
   background-color: #222222;
   min-height: 100vh;
   min-width: 100vw;
-  z-index: 9;
-  position:absolute;
+  z-index: 10x;
+  position:fixed;
   left: 0;
   top: 0;
 
